src/azure-cli-core/azure/cli/core/auth/identity.py

```diff
@@ -346,3 +353,23 @@ def _try_remove(path):
         os.remove(path)
     except FileNotFoundError:
         pass
+
+
+def get_environment_credential():
+    # A temporary workaround used by rdbms module to use environment credential.
+    # TODO: Integrate with Identity and utilize MSAL HTTP and token cache to officially implement
+    #  https://github.com/Azure/azure-cli/issues/10241
+    from os import getenv
+
+    sp_auth = ServicePrincipalAuth({
+        _TENANT: getenv(AZURE_TENANT_ID),
+        _CLIENT_ID: getenv(AZURE_CLIENT_ID),
+        _CLIENT_SECRET: getenv(AZURE_CLIENT_SECRET)
+    })
+
+    authority, _ = _get_authority_url(
+        # Override authority host if defined as env var
+        getenv(AZURE_AUTHORITY_HOST) or 'https://login.microsoftonline.com',
+        getenv(AZURE_TENANT_ID))
+    credentials = ServicePrincipalCredential(sp_auth, authority=authority)
```

Comment:
This line is broken by the `ServicePrincipalCredential`'s signature change in #29439.